crates/data_components/src/databricks/auth.rs

```diff
@@ -0,0 +1,194 @@
+/*
+Copyright 2024-2025 The Spice.ai OSS Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+     https://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+use async_trait::async_trait;
+use secrecy::{ExposeSecret, SecretString};
+use serde::{Deserialize, Serialize};
+use snafu::prelude::*;
+use std::time::Duration;
+use std::{fmt, sync::Arc};
+use tokio::{sync::watch, task::JoinHandle, time::sleep};
+use util::fibonacci_backoff::FibonacciBackoffBuilder;
+
+use crate::token_provider::{Result, TokenProvider};
+
+const TOKEN_REFRESH_BUFFER_SECS: u64 = 300;
+
+#[derive(Debug, Snafu)]
+pub enum Error {
+    #[snafu(display(
+        "Failed to obtain Databricks service principal token for machine-to-machine authentication."
+    ))]
+    UnableToGetToken { source: Box<dyn std::error::Error> },
+}
+
+#[derive(Clone)]
+pub struct DatabricksM2MTokenProvider {
+    endpoint: String,
+    client_id: String,
+
+    #[allow(dead_code)]
```

Comment:
Is this still `dead_code` now that they get used in `impl TokenProvider for DatabricksM2MTokenProvider`

Reply:
Right, that was from old implementation, not needed anymore